applications/StructuralMechanicsApplication/python_scripts/structural_response.py

```diff
@@ -433,9 +435,40 @@ def _GetAdjointParameters(self):
             with open(self.response_settings["adjoint_settings"].GetString(),'r') as parameter_file:
                 adjoint_parameters = Parameters( parameter_file.read() )
 
+            # prevent the case that primal solution is transfered by python and the HDF5 process
+            if self.use_python_primal_solution_data_transfer:
+                if adjoint_parameters["processes"].Has("list_other_processes"):
+                    for i in range(0,adjoint_parameters["processes"]["list_other_processes"].size()):
+                        process = adjoint_parameters["processes"]["list_other_processes"][i]
+                        if process["python_module"].GetString() ==  "single_mesh_temporal_input_process":
+                            raise Exception("For transfering the primal results 'python' is chosen but also 'single_mesh_temporal_input_process' is provided within the adjoint parameters!")
+
         return adjoint_parameters
 
 
+    def _UsePythonPrimalDataTransfer(self):
+        use_python_data_transfer = True
+
+        if self.response_settings["adjoint_settings"].GetString() == "auto":
+            if self.response_settings.Has("primal_data_transfer"):
+                primal_data_transfer = self.response_settings["primal_data_transfer"].GetString()
+                if primal_data_transfer != "python":
+                    raise Exception("Auto setup of adjoint parameters does not support {} for primal data transfer. Only available for 'python'.".format(primal_data_transfer))
+        else:
+            if self.response_settings.Has("primal_data_transfer"):
+                primal_data_transfer = self.response_settings["primal_data_transfer"].GetString()
+                if primal_data_transfer == "hdf5":
```

Comment:
is there another option than python and hdf5 planned? If not, i would make it a simple boolean flag, e.g. `primal_data_transfer_with_python: true`

Also i would make the setting mandatory, and not check if it is there or not. Or at least not dependent on the "auto" or not setting.